### ISSUE TITLE
Android (Mapbox 7.0.0-alpha.3) - City and state names flickering

### ISSUE THREAD

**Issue by Mordag** [mapbox/mapbox-gl-native#13506]
After updating to mapbox 7.0.0-alpha.3, city and state names are flickering every time the camera is moving. In general, the camera and map rendering performance got much worse with 7.0.0-alpha.3.

### Configuration
**Mapbox SDK versions:** 7.0.0-alpha.3

**Comment by ac87** [comment]
Seeing the same thing, I don't need to move the map to make things flicker as there is a 'my location' overlay moving an icon constantly which is causing the same refresh I guess.

The yellow circle which is also flashing, is an item on a symbol layer.
The circle and drawn line which aren't flashing are annotations.

![device-2018-12-05-080801](https://user-images.githubusercontent.com/20557626/49499344-a2b5aa00-f865-11e8-969d-d5c117fb5d22.gif)
The blur visible is just a GIF artifact, black border courtesy of Android Studio's screen recorder.

**Comment by LukasPaczos** [comment]
Thanks for reporting the issue @Mordag @ac87. Would you be able to provide an example setup? I wasn't able to reproduce locally on `master` or `7.0.0-alpha.3`.

**Comment by ac87** [comment]
Difficult to create a test without creating a new project. So I commented code until it behaved. 

I've tracked it down in my case to setting the Transition Duration to 0

```
mapView.getMapAsync(mapboxMap -> {
                ...
                mapboxMap.setTransitionDuration(0);
                ...
}
```

Interestingly if I force the style to change it behaves even with this code, so its only on the first style load.

**Comment by LukasPaczos** [comment]
This looks like the key, I was able to reproduce, thank you @ac87!

**Comment by LukasPaczos** [comment]
Bisecting points to https://github.com/mapbox/mapbox-gl-native/pull/13035/commits/78598ac19dbebe5512c3367650d8021144a8c83b.

/cc @brunoabinader @ChrisLoer

**Comment by ChrisLoer** [comment]
@brunoabinader Looks like a divide-by-zero issue:

https://github.com/mapbox/mapbox-gl-native/commit/78598ac19dbebe5512c3367650d8021144a8c83b#diff-6989ce9a7567f8da5dd71c1ca97c7491R249

**Comment by ChrisLoer** [comment]
Should be fixed by #13535.
